src/lib/Rating/index.tsx

```diff
@@ -0,0 +1,403 @@
+    newHover = clamp(newHover, precision, max)
+
+    setState((prev) =>
+      prev.hover === newHover && prev.focus === newHover
+        ? prev
+        : {
+            hover: newHover,
+            focus: newHover
+          }
+    )
+
+    setFocusVisible(false)
+
+    if (onChangeActive && hover !== newHover) {
+      onChangeActive(event, newHover)
+    }
+  }
+
+  const handleMouseLeave = (event: any) => {
+    if (onMouseLeave) {
+      onMouseLeave(event)
+    }
+
+    const newHover = -1
+    setState({
+      hover: newHover,
+      focus: newHover
+    })
+
+    if (onChangeActive && hover !== newHover) {
+      onChangeActive(event, newHover)
+    }
+  }
+
+  const handleChange = (event: any) => {
+    const newValue = parseFloat(event.target.value)
+
+    setValueState(newValue)
+
+    if (onChange) {
+      onChange(event, newValue)
+    }
+  }
+
+  const handleClear = (event: any) => {
+    // Ignore keyboard events
+    // https://github.com/facebook/react/issues/7407
+    if (event.clientX === 0 && event.clientY === 0) {
+      return
+    }
+
+    setState({
+      hover: -1,
+      focus: -1
+    })
+
+    setValueState(null)
+
+    if (onChange && parseFloat(event.target.value) === valueRounded) {
+      onChange(event, null)
+    }
+  }
+
+  const handleFocus = (event: any) => {
+    handleFocusVisible(event)
+    if (isFocusVisibleRef.current === true) {
+      setFocusVisible(true)
+    }
+
+    const newFocus = parseFloat(event.target.value)
+    setState((prev) => ({
+      hover: prev.hover,
+      focus: newFocus
+    }))
+
+    if (onChangeActive && focus !== newFocus) {
+      onChangeActive(event, newFocus)
+    }
+  }
+
+  const handleBlur = (event: any) => {
+    if (hover !== -1) {
+      return
+    }
+
+    handleBlurVisible()
+    if (isFocusVisibleRef.current === false) {
+      setFocusVisible(false)
+    }
+
+    const newFocus = -1
+    setState((prev) => ({
+      hover: prev.hover,
+      focus: newFocus
+    }))
+
+    if (onChangeActive && focus !== newFocus) {
+      onChangeActive(event, newFocus)
+    }
+  }
+
+  const item = (state: any, labelProps = {}) => {
+    const id = `${name}-${String(state.value).replace('.', '-')}`
+    const container = (
+      <span
+        className={clsx(styles.icon, {
+          [styles.iconEmpty]: !state.filled,
+          [styles.iconActive]: state.active
+        })}
+      >
+        <Star />
+      </span>
+    )
+
+    if (readOnly) {
+      return (
+        <span key={state.value} {...labelProps}>
+          {container}
+        </span>
+      )
+    }
+
+    return (
+      <React.Fragment key={state.value}>
+        <label className={styles.label} htmlFor={id} {...labelProps}>
+          {container}
+          <span className={styles.visuallyHidden}>{getLabelText(state.value)}</span>
+        </label>
+        <input
+          onFocus={handleFocus}
+          onBlur={handleBlur}
+          onChange={handleChange}
+          onClick={handleClear}
```

Comment:
need to also handle a click with enter key to clear, hopefully the key click will just change the checked state below and screen reader will announce.